### PR TITLE
DT macros.bnf fixes

### PR DESCRIPTION
**Pull request by mbolivar-nordic**
Mostly docs improvements and cosmetic fixes.

One true error in the ABNF itself (a missing underscore).

### CODE REVIEW

**Comment by mbolivar-nordic** [comment]
@jettr since I referred you to this file, I thought I'd give it a once over while it's on my mind.

File: doc/guides/dts/macros.bnf [hunk 1]
```diff
@@ -39,16 +39,20 @@ node-macro =/ %s"DT_N" path-id %s"_IRQ_IDX_" DIGIT
...
-; For fixed-partitions give a unique ordinal value for each partition */
+; Subnodes of the fixed-partitions compatible get macros which contain
+; a unique ordinal value for each partition
 node-macro =/ %s"DT_N" path-id %s"_PARTITION_ID" DIGIT
-; Macros are generated for each of the node's compatibles.
+; Macros are generated for each of a node's compatibles;
+; dt-name in this case is something like "vnd_device".
 node-macro =/ %s"DT_N" path-id %s"_COMPAT_MATCHES_" dt-name
-; The node identifier for the node's parent in the devicetree.
+; Every node gets one of these macros which expands to the node identifier
```
Comment (pabigot) [inline doc/guides/dts/macros.bnf:48]: except the root node, if we want to get picky.  Maybe it's not a node.

File: doc/guides/dts/macros.bnf [hunk 2]
```diff
@@ -39,16 +39,20 @@ node-macro =/ %s"DT_N" path-id %s"_IRQ_IDX_" DIGIT
...
+; for that node's parent in the devicetree.
 node-macro =/ %s"DT_N" path-id %s"_PARENT"
 ; These are used internally by DT_FOREACH_CHILD, which iterates over
 ; each child node.
 node-macro =/ %s"DT_N" path-id %s"_FOREACH_CHILD"
-; The node's status macro
+; The node's status macro; dt-name in this case is someting like "okay"
```
Comment (pabigot) [inline doc/guides/dts/macros.bnf:54]: s/someting/something/

**Review by pabigot — CHANGES_REQUESTED** [review]
one typo.

File: doc/guides/dts/macros.bnf (hunk 1 above)
Comment (mbolivar-nordic) [reply doc/guides/dts/macros.bnf:48]: It's a node and this is an ABNF, so we should definitely be picky :)

**Review by pabigot — APPROVED** [review]

**Review by galak — APPROVED** [review]

**Comment by jettr** [comment]
> @jettr since I referred you to this file, I thought I'd give it a once over while it's on my mind.

I'm slow the the review since it already merged, but it looks good to me!

**Pull request merged**